dev/dev-env-setup.sh

```diff
@@ -0,0 +1,254 @@
+pyenv_exist=$(command -v pyenv)
+
+if [ -z "$pyenv_exist" ]; then
+  if [ -z "$MLFLOW_DEV_ENV_PYENV_INSTALL" ]; then
+    read -p "pyenv is required to be installed to manage python versions. Would you like to install it? $(tput bold)(y/n)$(tput sgr0): " -n 1 -r
+    echo
+  fi
+  if [[ $REPLY =~ ^[Yy]$ || $MLFLOW_DEV_ENV_PYENV_INSTALL == 1 ]]; then
+    if [[ "$machine" == mac ]]; then
+      # Check if brew is installed and install it if it isn't present
+      # Note: if xcode isn't installed, this will fail.
+      if [ -z "$(command -v brew)" ]; then
+        echo "Brew is required to install pyenv on MacOS. Installing in your home directory."
+        wget -O ~/brew_install.sh https://raw.githubusercontent.com/Homebrew/install/master/install.sh
+        bash ~/brew_install.sh
+      fi
+      echo "Updating brew and installing pyenv..."
+      echo "Note: this will probably take a considerable amount of time."
+      brew update
+      brew install pyenv
+    elif [[ "$machine" == linux ]]; then
+      sudo apt-get update -y
+      sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
+      libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+      libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+      # Install pyenv from source
+      git clone --depth 1 https://github.com/pyenv/pyenv.git "$HOME/.pyenv"
+      PYENV_ROOT="$HOME/.pyenv"
+      PYENV_BIN="$PYENV_ROOT/bin"
+      if [ "$MLFLOW_DEV_ENV_CI_RUN" == 1 ]; then
+        echo "$PYENV_BIN" >> "$GITHUB_PATH"
+        echo "PYENV_ROOT=$PYENV_ROOT" >> "$GITHUB_ENV"
+      fi
+    elif [[ "$machine" == win ]]; then
+      if [ -z "$(command -v pip)" ]; then
+        echo "A pip installation cannot be found. Install pip first."
+        exit 1
+      fi
+      # install via system pip as per pyenv-win docs
+      pip install $(quietpip) pyenv-win --target "$HOME\\.pyenv"
+      if [ "$MLFLOW_DEV_ENV_CI_RUN" == 1 ]; then
+        {
+          echo "PYENV=$USERPROFILE\.pyenv\pyenv-win\\";
+          echo "PYENV_ROOT=$USERPROFILE\.pyenv\pyenv-win\\";
+          echo "PYENV_HOME=$USERPROFILE\.pyenv\pyenv-win\\";
+        } >> "$GITHUB_ENV"
+        echo "$USERPROFILE\.pyenv\pyenv-win\\bin\\" >> "$GITHUB_PATH"
+      fi
+    else
+      echo "Unknown operating system environment: $machine exiting."
+      exit 1
+    fi
+  else
+    if [[ "$machine" == win ]]; then
+      PYENV_README=https://github.com/pyenv-win/pyenv-win/blob/master/README.md
+    elif [[ "$machine" == mac || "$machine" == linux ]]; then
+      PYENV_README=https://github.com/pyenv/pyenv/blob/master/README.md
+    else
+      echo "The current OS is unknown. Please visit: https://github.com/pyenv/pyenv#installation for instructions. "
+      exit 1
+    fi
+    echo "pyenv is required to use this environment setup script. Please install by following instructions here: $PYENV_README"
+    exit 1
+  fi
+fi
+
+MLFLOW_HOME=$(pwd)
```

Comment:
```suggestion
REPO_ROOT=$(git rev-parse --show-toplevel)
```

might make more sense.